x/ibc/core/03-connection/keeper/keeper.go

```diff
@@ -105,6 +115,24 @@ func (k Keeper) SetClientConnectionPaths(ctx sdk.Context, clientID string, paths
 	store.Set(host.ClientConnectionsKey(clientID), bz)
 }
 
+// GetNextConnectionSequence gets the next connection sequence from the store.
+func (k Keeper) GetNextConnectionSequence(ctx sdk.Context) uint64 {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get([]byte(types.KeyNextConnectionSequence))
+	if bz == nil {
+		panic("next connection sequence is nil")
```

Comment:
why not return `0, error`?

Reply:
this means we didn't set it on init genesis. It is an assumption that the global counter is always set